src/main/java/folk/sisby/tinkerers_smithing/mixin/MixinRegistrySyncManager.java

```diff
@@ -0,0 +1,29 @@
+package folk.sisby.tinkerers_smithing.mixin;
+
+import folk.sisby.tinkerers_smithing.TinkerersSmithing;
+import net.fabricmc.fabric.impl.registry.sync.RegistrySyncManager;
+import net.minecraft.registry.Registry;
+import net.minecraft.util.Identifier;
+import org.jetbrains.annotations.Nullable;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import java.util.Objects;
+
+@Mixin(RegistrySyncManager.class)
+public abstract class MixinRegistrySyncManager {
```

Comment:
TSmithing on other versions has reasonable EMI support both with and without TSmithing itself on the client - can you check if this preserves that?

Reply:
I was able to write a new version of the mixin: 
```java
@ModifyVariable(method = "createAndPopulateRegistryMap", at = @At("STORE"), ordinal = 1, remap = false)
private static Identifier skipRecipes(Identifier id, @Local(ordinal = 0) Object obj) {
	if (ServerRecipe.class.isAssignableFrom(obj.getClass().getEnclosingClass())) {
		return null;
	}
	return id;
}
```
Which I think is quite clean, and works, however it also completely breaks EMI compat, somehow an unexplained internal NullPointerException gets thrown *somewhere* with this version of the mixin. The original mixin works fine.

Reply:
I was being silly, just had to check for nulls in the mixin itself.

Reply:
Noticeable issues (with TSmithing only serverside, no EMI on server because it doesn't seem to work on the server currently):
- EMI thinks netherite is repaired with netherite (it cannot be, and without TSmithing on client the anvil will *very* briefly think it can)
- EMI doesn't suggest that netherite scrap can be used to dework
Other than that things seem to be working fine?

With TSmithing clientside as well, these issues are gone.

Reply:
pretty sure that's the expected level of parity! great work ^^